### ISSUE TITLE
Images server (AWS S3) is not secure enough for iOS9

### ISSUE THREAD

**Issue by murraycu** [zooniverse/Galaxy-Zoo#228]
This is really an issue for the people running the servers rather than the people developing the web client code. However:

The AWS S3 server's SSL/TLS support is apparently not as good as it could be.

This is a (slight) problem for app code running on iOS9, such as my ios-galaxyzoo app in progress. It's apparently a known problem with AWS S3, and it's apparently fairly easy to fix on the servers:
http://stackoverflow.com/questions/32500655/ios-9-app-download-from-amazon-s3-ssl-error-tls-1-2-support

Even though I can workaround it in my iOS app, it seems like something that should be fixed for general security reasons.

**Comment by willettk** [comment]
@astopy?

**Comment by murraycu** [comment]
Actually, I now see that the main problem for iOS is that the image URLs really are http rather than https. For instance:
http://www.galaxyzoo.org.s3.amazonaws.com/subjects/decals/standard/J214413.38-005656.5_standard.jpeg

They can be served via https just by changing the protocol in the URL:
https://www.galaxyzoo.org.s3.amazonaws.com/subjects/decals/standard/J214413.38-005656.5_standard.jpeg

But that's not going to work as long as the certificate is self-signed. It might also have the amazon S3 SSL/TLS problem anyway.

**Comment by astopy** [comment]
It’s not that the cert is self-signed, it’s just not valid for that domain name. If you can transform the URLs to the following format then you should be able to access them without any certificate errors:

https://s3.amazonaws.com/www.galaxyzoo.org/subjects/decals/standard/J214413.38-005656.5_standard.jpeg

**Comment by murraycu** [comment]
Thanks.

Then ideally the server should provide those URLs via the REST queries, I guess. For instance, in results for queries such as this:
https://api.zooniverse.org/projects/galaxy_zoo/groups/55db71251766276613000001/subjects?limit=4

Then it would be fixed for all clients.

The server code is not open source, I believe, so not something that I can fix here on github.

**Comment by astopy** [comment]
Unfortunately those URLs are hard-coded in the MongoDB records, so it’s not something we can easily mass-update.

**Comment by murraycu** [comment]
How are those URLs generated? Maybe this can be corrected when the database records are created for the next survey subjects.

**Comment by astopy** [comment]
Yes, agreed. They’re generated by a script which creates the records, so we can set them differently for any future subjects.

**Comment by murraycu** [comment]
Is this script in github somewhere?

**Comment by astopy** [comment]
No, I don’t think it’s available anywhere unfortunately.
